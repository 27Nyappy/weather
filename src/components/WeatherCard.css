.Card {
  @media only screen and (max-width: 430px) {
    width: 300px;
  }

  @media screen and (min-width: 431px) and (max-width: 600px) {
    width: 440px;
  }

  @media screen and (min-width: 601px) and (max-width: 768px) {
    width: 520px;
  }

  @media screen and (min-width: 769px) and (max-width: 834px) {
    width: 540px;
  }

  @media screen and (min-width: 835px) {
    width: 560px;
  }

  text-align: center;

  h2 {
    font-size: 20pt;
    font-weight: 400;
    margin-bottom: 10px;
  }

  p:first-of-type {
    width: 90%;
    font-size: 10pt;
    white-space: normal;
    
  }

  div {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    img {
      width: 40%;
      max-width: 150px;
    }

    span {
      font-size: 20pt;
      font-weight: 400;
    }
  }

}