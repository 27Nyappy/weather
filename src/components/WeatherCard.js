import React from 'react';
import PropTypes from 'prop-types';
import styles from './WeatherCard.css';

const WeatherCard = ({ mainTemp, feelsLike, maxTemp, minTemp, humidity, detail, description, icon, date, city }) => {
  return (
    <div className={styles.Card}>
      <h2>{city}</h2>
      <p>{date}</p>
      <div>
        <span>{mainTemp}&#176;</span>
        <img src={icon} alt="Weather Icon" />
      </div>
      <p>{maxTemp}&#176;/{minTemp}&#176; Feels like {feelsLike}&#176;</p>
      <p>{detail}, {description} and {humidity}% humidity</p>
    </div>
  );
};

WeatherCard.propTypes = {
  mainTemp: PropTypes.number.isRequired,
  feelsLike: PropTypes.number.isRequired,
  maxTemp: PropTypes.number.isRequired,
  minTemp: PropTypes.number.isRequired,
  humidity: PropTypes.number.isRequired,
  detail: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired
};

export default WeatherCard;
