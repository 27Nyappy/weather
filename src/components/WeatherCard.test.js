import React from 'react';
import { shallow } from 'enzyme';
import WeatherCard from './WeatherCard';


describe('Weather Card Component', () => {
  const testCard = {
    mainTemp: 50.13,
    feelsLike: 43.68,
    maxTemp: 53.6,
    minTemp: 46,
    humidity: 81,
    detail: 'Rain',
    description: 'moderate rain',
    icon: 'http://openweathermap.org/img/wn/10n@2x.png',
    date: 'Mon Jan 27 2020 22:57:33 GMT-0800 (Pacific Standard Time)',
    city: 'Portland'
  };

  it('Renders Weather Card', () => {
    const wrapper = shallow(<WeatherCard mainTemp={testCard.mainTemp} feelsLike={testCard.feelsLike} maxTemp={testCard.maxTemp} minTemp={testCard.minTemp} humidity={testCard.humidity} detail={testCard.detail} description={testCard.description} icon={testCard.icon} date={testCard.date} city={testCard.city} />);
    expect(wrapper).toMatchSnapshot();
  });
});