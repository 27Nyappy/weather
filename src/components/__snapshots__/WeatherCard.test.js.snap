// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Card Component Renders Weather Card 1`] = `
<div
  className="Card"
>
  <h2>
    Portland
  </h2>
  <p>
    Mon Jan 27 2020 22:57:33 GMT-0800 (Pacific Standard Time)
  </p>
  <div>
    <span>
      50.13
      째
    </span>
    <img
      alt="Weather Icon"
      src="http://openweathermap.org/img/wn/10n@2x.png"
    />
  </div>
  <p>
    53.6
    째/
    46
    째 Feels like 
    43.68
    째
  </p>
  <p>
    Rain
    , 
    moderate rain
     and 
    81
    % humidity
  </p>
</div>
`;
